src/request/import/ImportFile.js

```diff
@@ -118,50 +142,30 @@ class ImportFile {
     }
 
     /**
-     * TODO LoginFile
-     * @param {string?} passphrase
-     * @returns {Promise<?Key>}
+     * @param {string} passphrase
+     * @returns {Promise<Key?>}
      */
     async _decryptAndStoreKey(passphrase) {
         TopLevelApi.setLoading(true);
         try {
-            // Separating the processing of the encryptionKey (password) and the secret (key) is necessary
-            // to cover these scenarios:
-            //     1. Encrypted key file with password or PIN
-            //     2. Unencrypted key file and no new password set
-            //     3. Unencrypted key file and new password set
-
-            /** @type {Nimiq.Entropy | Nimiq.PrivateKey} */
-            let secret;
-            let encryptionKey = null;
-
-            if (passphrase !== null) {
-                encryptionKey = Utf8Tools.stringToUtf8ByteArray(passphrase);
-            }
-
-            if (this._encryptedKey.byteLength === KeyStore.ENCRYPTED_SECRET_SIZE_V2
-                || this._encryptedKey.byteLength === KeyStore.ENCRYPTED_SECRET_SIZE) {
-                // Make sure read position is at 0 even after a wrong passphrase
-                this._encryptedKey.reset();
-
-                secret = await Nimiq.Secret.fromEncrypted(
-                    this._encryptedKey,
-                    /** @type {Uint8Array} */ (encryptionKey),
-                );
-            } else {
-                // Key File was not encrypted and the imported Uint8Array is the plain secret
-                secret = new Nimiq.Entropy(this._encryptedKey);
-            }
-
-            const key = new Key(secret, false);
-
-            await KeyStore.instance.put(key, encryptionKey || undefined);
+            const encryptionKey = Utf8Tools.stringToUtf8ByteArray(passphrase);
+
+            // Make sure read position is at 0 even after a wrong passphrase
+            this._encryptedKey.reset();
+
+            const secret = await Nimiq.Secret.fromEncrypted(this._encryptedKey, encryptionKey);
+
+            // If this code runs, the password was correct
+            /** @type {HTMLElement} */
+            (this.$unlockAccountPage.querySelector('.lock-locked')).classList.add('lock-unlocked');
 
+            const key = new Key(secret, this._flags.hasPin);
+            await KeyStore.instance.put(key, encryptionKey);
             return key;
         } catch (event) {
             console.error(event);
             TopLevelApi.setLoading(false);
```

Comment:
Why is this line in here?

Reply:
The `setLoading`? Because there is a `setLoading(true)` above, that displays the loading indicator for the duration that it takes to decode the encrypted key. If the password is wrong, the `catch` block is executed and we want the loading overlay to go away again, so the user can try again.